horovod/keras/callbacks.py

```diff
@@ -152,3 +152,24 @@ def __init__(self, warmup_epochs=5, momentum_correction=True, steps_per_epoch=No
         """
         super(LearningRateWarmupCallback, self).__init__(K, warmup_epochs, momentum_correction,
                                                          steps_per_epoch, verbose, initial_lr)
+
+
+class ReturnedModelCheckpoint(keras.callbacks.ModelCheckpoint):
+    def __init__(self, filepath=None,
+                 monitor='val_loss',
+                 verbose=0,
+                 save_best_only=True,
```

Comment:
Is `save_best_only` needed here?  It seems that with a specific file path provided, it always saves the best only.

Reply:
Without `save_best_only`, in every epoch or whatever the monitoring interval is it overwrites the previous checkpoint. With it, it will only overwrite if the current checkpoint has better metric than the previously saved checkpoint. 

I think we need it.

Reply:
Makes sense, but is there ever a case where the user would want to set `save_best_only=False` using this class?

Reply:
I cannot think of a use case off the top of my head. But I think, it is a better choice to conform with the defaults of the `ModelCheckpoint` args to avoid confusion for the users.

Reply:
I am open to change this if you have a strong preference.

Reply:
My concern is that if we expose `save_best_only` to the user, but it doesn't do anything, that would create even more confusion for the user.  I think for now we may want to remove it, and always set `save_best_only=True` when initializing the parent constructor.